### PR TITLE
Feat/sharepoint

### PR DESCRIPTION
**Pull request by laurentleseigneur**
feat(document): add support for binary body content

* update definition to 1.3.1
* change body param type from `String` to `Serializable`
* add support for document (Bonita version) using byte array as payload
* add `application/octet-stream` content type for PUT and POST
* add spotless maven plugin to fix formatting issues between IDEs
* fix gihub action in order to make then working on forked repo, skip steps with required secrets (SONAR)
* add maven cache support

### CODE REVIEW

**Comment by laurentleseigneur** [comment]
@rbioteau can you review this please ? thanks

**Comment by laurentleseigneur** [comment]
need to update sonar skip condition, or is it just that PR from fork don't share GH secrets ?

File: pom.xml [hunk 1]
```diff
@@ -329,6 +329,18 @@
 					</execution>
 				</executions>
 			</plugin>
+			<plugin>
```
Comment (rbioteau) [inline pom.xml]: Not relevant with actual PR

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -16,16 +16,39 @@ jobs:
...
 
     - name: Build and test
       env:
         GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
         SONAR_TOKEN: ${{secrets.SONAR_TOKEN}}
-      run: ./mvnw -B clean verify sonar:sonar -Dsonar.login=${{ secrets.SONAR_TOKEN }}
-    - run: mkdir staging && cp target/*.zip staging
+      run: ./mvnw -B clean spotless:check verify
+
+    - name: Check whether SONAR_TOKEN is set
```
Comment (rbioteau) [inline .github/workflows/build.yml:31]: 👍

File: .github/workflows/commit-message-check.yml [hunk 3]
```diff
@@ -9,6 +9,7 @@ on:
 jobs:
   check-commit-message:
     name: Check Commit Message
+    if: github.repository_owner == 'bonitasoft'
```
Comment (rbioteau) [inline .github/workflows/commit-message-check.yml]: why ?

File: src/main/resources-filtered/rest-put.def [hunk 4]
```diff
@@ -48,8 +48,10 @@
     <widget xsi:type="definition:Text" id="urlWidget" inputName="url"/>
     <widget xsi:type="definition:Select" id="contentTypeWidget" inputName="contentType">
 		<items>application/json</items>
+		<items>application/octet-stream</items>
 		<items>application/xml</items>
 		<items>application/x-www-form-urlencoded</items>
+		<items>application/octet-stream</items>
```
Comment (rbioteau) [inline src/main/resources-filtered/rest-put.def]: duplicated

**Review by rbioteau — CHANGES_REQUESTED** [review]

**Review by rbioteau — APPROVED** [review]

**Pull request merged**